### PR TITLE
Simplify and cleanup RgbPwmLed + extension

### PR DESCRIPTION
**Pull request by adrianstevens**

### CODE REVIEW

File: Source/Meadow.Foundation.Core/Leds/RgbPwmLedExtensions.cs [hunk 1]
```diff
@@ -1,156 +1,191 @@
...
+
+            await StopAnimation(led);
+
+            lock (syncRoot)
             {
-                led.Stop();
+                cancellationTokenSource = new CancellationTokenSource();
 
-                var animationTask = new Thread((s) =>
+                animationTask = new Task(() =>
```
Comment (ctacke) [inline Source/Meadow.Foundation.Core/Leds/RgbPwmLedExtensions.cs:103]: What if I want to have > 1 blinking LED?  You need to track all animation Tasks, not just the most recent, yes?

File: Source/Meadow.Foundation.Core.Samples/Leds.LedBarGraph_Sample/MeadowApp.cs [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 namespace Leds.LedBarGraph_Sample
 {
-    public class MeadowApp : App<F7FeatherV2>
+    public class MeadowApp : App<F7FeatherV1>
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Core.Samples/Leds.LedBarGraph_Sample/MeadowApp.cs:10]: FeatherV1

File: Source/Meadow.Foundation.Core.Samples/Leds.PwmLedBarGraph_Sample/MeadowApp.cs [hunk 3]
```diff
@@ -2,14 +2,13 @@
...
 using Meadow.Hardware;
-using Meadow.Peripherals.Leds;
 using Meadow.Units;
 using System;
 using System.Threading.Tasks;
 
 namespace Leds.PwmLedBarGraph_Sample
 {
-    public class MeadowApp : App<F7FeatherV2>
+    public class MeadowApp : App<F7FeatherV1>
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Core.Samples/Leds.PwmLedBarGraph_Sample/MeadowApp.cs:11]: FeatherV1

File: Source/Meadow.Foundation.Core.Samples/Leds.LedBarGraph_Sample/MeadowApp.cs (hunk 2 above)
Comment (jorgedevs) [reply Source/Meadow.Foundation.Core.Samples/Leds.LedBarGraph_Sample/MeadowApp.cs]: Fixed

File: Source/Meadow.Foundation.Core.Samples/Leds.RgbPwmLed_Onboard_Sample/MeadowApp.cs [hunk 4]
```diff
@@ -18,48 +17,47 @@ public class MeadowApp : App<F7FeatherV2>
...
                 Device.Pins.OnboardLedRed,
                 Device.Pins.OnboardLedGreen,
-                Device.Pins.OnboardLedBlue,
-                commonType: CommonType.CommonAnode);
+                Device.Pins.OnboardLedBlue);
 
             return Task.CompletedTask;
         }
 
         public override Task Run()
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Core.Samples/Leds.RgbPwmLed_Onboard_Sample/MeadowApp.cs:29]: You should be able to directly return TestColors - I think I did this when we were fighting to get it working

File: Source/Meadow.Foundation.Core.Samples/Leds.RgbPwmLed_Onboard_Sample/MeadowApp.cs [hunk 5]
```diff
@@ -18,48 +17,47 @@ public class MeadowApp : App<F7FeatherV2>
...
-            Thread.Sleep(3000);
-            onboardLed.SetColor(Color.MediumPurple);
-            Thread.Sleep(3000);
-            onboardLed.SetColor(Color.FromHex("#23abe3"));
-        }
+            while (true)
+            {
+                Console.WriteLine("SetColor(RgbLedColors.Red);");
+                onboardLed.SetColor(Color.Red);
+                Thread.Sleep(3000);
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Core.Samples/Leds.RgbPwmLed_Onboard_Sample/MeadowApp.cs:42]: And these could be await Task.Delay .... (not a big deal)

File: Source/Meadow.Foundation.Core/Leds/Led.Extensions.cs [hunk 6]
```diff
@@ -0,0 +1,69 @@
+﻿using System;
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Core/Leds/Led.Extensions.cs:1]: Let's rename this file to something like Led.Animations.cs so it's not an extension anymore

File: Source/Meadow.Foundation.Core/Leds/LedBarGraph.Extensions.cs [hunk 7]
```diff
@@ -0,0 +1,124 @@
+﻿using System;
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Core/Leds/LedBarGraph.Extensions.cs:1]: And here - let's renaming to LedBarGraph.Animations.cs

File: Source/Meadow.Foundation.Core/Leds/LedBarGraph.cs [hunk 8]
```diff
@@ -66,177 +57,44 @@ public LedBarGraph(IDigitalOutputPort[] ports)
...
+
             var value = percentage * Count;
             value += 0.5f;
 
             for (int i = 1; i <= Count; i++)
             {
                 if (i <= value)
                 {
-                    SetLed(i - 1, true);
+                    await SetLed(i - 1, true);
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Core/Leds/LedBarGraph.cs:91]: You could get fancy and change to 

await SetLed(i -1, i <= value);

File: Source/Meadow.Foundation.Core/Leds/PwmLed.Extensions.cs [hunk 9]
```diff
@@ -0,0 +1,173 @@
+﻿using System;
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Core/Leds/PwmLed.Extensions.cs:1]: PwmLed.Animations.cs

File: Source/Meadow.Foundation.Core/Leds/RgbLed.Extensions.cs [hunk 10]
```diff
@@ -0,0 +1,95 @@
+﻿using System;
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Core/Leds/RgbLed.Extensions.cs:1]: RgbPwm.Animations.cs

File: Source/Meadow.Foundation.Core/Leds/RgbLed.cs [hunk 11]
```diff
@@ -47,14 +42,14 @@ public bool IsOn
             get => isOn;
             set
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Core/Leds/RgbLed.cs:43]: set => UpdateLed(isOn = value);

File: Source/Meadow.Foundation.Core/Leds/RgbPwmLed.Extensions.cs [hunk 12]
```diff
@@ -0,0 +1,248 @@
+﻿using System;
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Core/Leds/RgbPwmLed.Extensions.cs:1]: RgbPwmLed.Animations.cs

**Review by jorgedevs — APPROVED** [review]

**Pull request merged**